tests/unit/compute-javascript-benefits.spec.ts

```diff
@@ -233,3 +233,83 @@ describe("computeAides", function () {
     ).toBe(0)
   })
 })
+
+describe("Test condition taux_incapacite", function () {
+
+  let situation_handicap
+  let benefit_situation_handicap
+
+  beforeEach(() => {
+    situation_handicap = {
+      demandeur: {
+        activite: "situation_handicap",
+        taux_incapacite: 30
+      },
+    }
+    benefit_situation_handicap = {
+      profils: [
+        {
+          type: "situation_handicap",
+          conditions: [{
+            type: "taux_incapacite",
+            value: 50,
+            operator: ">="
+          }]
+        }]
+    };
+  })
+
+  afterEach(() => {
+    situation_handicap = {
+      demandeur: {
+        activite: "situation_handicap",
+        taux_incapacite: 30
+      },
+    }
+    benefit_situation_handicap = {
+      profils: [
+        {
+          type: "situation_handicap",
+          conditions: [{
+            type: "taux_incapacite",
+            value: 50,
+            operator: ">="
+          }]
+        }]
+    };
+  })
```

Comment:
Tout ceci me paraît simplifiable :

```suggestion
  const situation_handicap = {
    demandeur: {
      activite: "situation_handicap",
      taux_incapacite: 30,
    },
  }
  const benefit_situation_handicap = {
    profils: [
      {
        type: "situation_handicap",
        conditions: [
          {
            type: "taux_incapacite",
            value: 50,
            operator: ">=",
          },
        ],
      },
    ],
  }
```

Reply:
Les valeurs de `situation_handicap` et `benefit_situation_handicap` sont modifiées à chaque tests (à part le premier).
Exemple : 
```python
  it("Checks that someone with incapacite 50 is not eligible with > 50", function () {
    situation_handicap.demandeur.taux_incapacite = 50
    benefit_situation_handicap.profils[0].conditions[0].operator = ">"

    expect(
      testProfileEligibility(benefit_situation_handicap, {
        situation: situation_handicap,
      })
    ).toBe(false)
  })
```
On pourrait en faire des constantes mais ça ferait beaucoup de bruits dans les tests.
Dans l'exemple ci-dessus, on met en valeur les propriétés testées.
Le `afterEach` me semble important pour éviter toute corrélation des tests.

Reply:
Ma suggestion supprime du code en double, du coup j'ai du mal à comprendre pourquoi tu penses que ça ferait du bruit ? 🤔

Ce qu'on peut faire si tu veux absolument réinitialiser tes variables à chaque test, c'est de mettre les déclarations uniquement dans le `beforeEach` plutôt que dans le `afterEach` car ça n'a pas trop de sens d'initialiser une variable après un test

Reply:
Pardon, je me suis un peu embrouillé.

J'ai supprimé le afterEach car il n'est effectivement pas utile.

Pour ce qui de l'affectation multiple des variables `situation_handicap` et `benefit_situation_handicap`, ce que je veux dire c'est qu'elles sont éditées dans les tests.
Si on les déclare comme constante une fois pour tous les tests, on ne pourra plus modifier les valeurs qui nous intéressent lors de chaque test.

Si l'on veut des constantes, il faudra donc redéclarer et initialiser les variables pour chaque test et c'est ça qui créerait du bruit.
Ce serait la méthode la plus sûre pour éviter les effets de bords mais en réaffectant ces deux variables avec les même valeurs, il me semble qu'on limtie l'effet de bord et on simplifie la lecture en évitant de dupliquer du code